src/components/Notifications/NotificationsLog/NotificationsLogDateFilter.tsx

```diff
@@ -0,0 +1,295 @@
+    () => (props.period[0] ? format(props.period[0], 'yyyy-MM-dd') : undefined),
+    [props.period]
+  );
+  const endValue: string | undefined = React.useMemo(
+    () => (props.period[1] ? format(props.period[1], 'yyyy-MM-dd') : undefined),
+    [props.period]
+  );
+
+  const startDateRef = useRef<DatePickerRef>(null);
+  const endDateRef = useRef<DatePickerRef>(null);
+
+  const onClickStartDateInput = React.useCallback(() => {
+    startDateRef.current?.setCalendarOpen(true);
+  }, [startDateRef]);
+
+  const onClickEndDateInput = React.useCallback(() => {
+    endDateRef.current?.setCalendarOpen(true);
+  }, [endDateRef]);
+
+  const startDateInputProps = React.useMemo<TextInputProps>(
+    () => ({
+      ...dateInputProps,
+      onClick: onClickStartDateInput,
+    }),
+    [onClickStartDateInput]
+  );
+
+  const endDateInputProps = React.useMemo<TextInputProps>(
+    () => ({
+      ...dateInputProps,
+      onClick: onClickEndDateInput,
+    }),
+    [onClickEndDateInput]
+  );
+
+  return (
+    <Split>
+      <SplitItem>
+        <DatePicker
+          placeholder="Start"
+          inputProps={startDateInputProps}
+          validators={startRangeValidators}
+          onChange={setStartDate}
+          value={startValue}
+          ref={startDateRef}
+        />
+      </SplitItem>
+      <SplitItem>
+        <Bullseye>
+          <span className={toClassName}>to</span>
+        </Bullseye>
+      </SplitItem>
+      <SplitItem>
+        <DatePicker
+          placeholder="End"
+          inputProps={endDateInputProps}
+          validators={endRangeValidators}
+          onChange={setEndDate}
+          value={endValue}
+          ref={endDateRef}
+        />
+      </SplitItem>
+    </Split>
+  );
+};
+
+export interface NotificationsLogDateFilterProps {
+  value: NotificationsLogDateFilterValue;
+  setValue: (value: NotificationsLogDateFilterValue) => void;
+  retentionDays: number;
+  period: EventPeriod;
+  setPeriod: Dispatch<SetStateAction<EventPeriod>>;
+}
+
+export const NotificationsLogDateFilter: React.FunctionComponent<NotificationsLogDateFilterProps> =
+  (props) => {
+    const options = React.useMemo(
+      () =>
+        Object.values(NotificationsLogDateFilterValue).map((v) => (
+          <SelectOption key={v} value={new EventLogSelectObject(v)} />
+        )),
+      []
+    );
+    const [isOpen, setOpen] = React.useState(false);
+    const onToggle = React.useCallback(
+      () => setOpen((prev) => !prev),
+      [setOpen]
+    );
+    const value = React.useMemo(
+      () => new EventLogSelectObject(props.value),
+      [props.value]
+    );
+    const onSelect = React.useCallback(
+      (_e: unknown, selectObject: SelectOptionObject | string) => {
+        const setValue = props.setValue;
+        if (selectObject instanceof EventLogSelectObject) {
+          setValue(selectObject.value);
+          setOpen(false);
+        }
+      },
+      [props.setValue]
+    );
```

Comment:
Since this is called just once I would consider moving this directly to `Select` component without the `useCallback` ptimization.

Reply:
Fixed